src/utilities/createRouter.ts

```diff
@@ -2,32 +2,56 @@ import { App } from 'vue'
 import { RouterLink, RouterView } from '@/components'
 import { routerInjectionKey, routerRejectionKey } from '@/compositions'
 import { Routes, Router, RouterOptions, RouterImplementation } from '@/types'
+import { RouterRejectionError } from '@/types/errors'
 import { createRouteMethods, createRouterNavigation, resolveRoutes } from '@/utilities'
 import { createRouterPush } from '@/utilities/createRouterPush'
-import { createRouterReject } from '@/utilities/createRouterReject'
+import { createRouterReject, notFoundRouteResolved } from '@/utilities/createRouterReject'
 import { createRouterReplace } from '@/utilities/createRouterReplace'
 import { createRouterResolve } from '@/utilities/createRouterResolve'
 import { createRouterRoute } from '@/utilities/createRouterRoute'
+import { getInitialUrl } from '@/utilities/getInitialUrl'
+import { routeMatch } from '@/utilities/routeMatch'
 
 export function createRouter<const T extends Routes>(routes: T, options: RouterOptions = {}): Router<T> {
   const resolved = resolveRoutes(routes)
   const resolve = createRouterResolve({ resolved })
 
+  const onLocationUpdate = async (url: string): Promise<void> => {
+    const matched = routeMatch(resolved, url)
+
+    if (!matched) {
+      await executeMiddleware(notFoundRouteResolved)
+      return reject('NotFound')
+    }
+
+    try {
+      await executeMiddleware(matched)
+      updateRoute(matched)
+      clearRejection()
+    } catch (error) {
+      if (error instanceof RouterRejectionError) {
+        reject(error.type)
+      }
+    }
```

Comment:
This isn't quite right. When a reject happens the route should still be updated. Which I think I had wrong in my PR as well. 

Also we cannot catch and silence every error. So we'll need to re throw if the error is not a specific instance we're expecting. I'm also thinking since we already have navigation here, we might as well just catch the middleware errors directly here and do navigation. Rather than passing navigation into createRouterRoute.

Reply:
I think you resolved this in your PR?